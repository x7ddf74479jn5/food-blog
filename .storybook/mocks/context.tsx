import { defaultMockRouter } from "mocks/next/router";
import { AppRouterContext } from "mocks/next/router";
import { SearchProvider } from "@/components/organisms/SearchArea/SearchContext";
import { createMockRouterHoc } from "mocks/next/router";
import React from "react";

export const withContext = (Story: React.ReactElement) => {
  return (
    <AppRouterContext.Provider value={defaultMockRouter}>
      <SearchProvider>{Story}</SearchProvider>
    </AppRouterContext.Provider>
  );
};

export const withRouterContext = createMockRouterHoc();
