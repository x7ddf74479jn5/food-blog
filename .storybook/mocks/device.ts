export const device = {
  desktop: "PC HD",
  mobile: "iPhone12",
  tablet: "iPad Pro 12.9-in",
};
