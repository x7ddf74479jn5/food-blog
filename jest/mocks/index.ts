export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
