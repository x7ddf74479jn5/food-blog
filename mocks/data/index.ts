export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
