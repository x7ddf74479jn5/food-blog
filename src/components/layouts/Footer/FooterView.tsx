import { SiteTitle } from "@/components/atoms/SiteTitle";

type Props = {
  organization: string;
};

export const FooterView: React.FC<Props> = ({ organization }) => (
  <footer className="mx-auto mt-auto flex w-full flex-row place-content-center gap-2 bg-gray-50 py-2 text-xs dark:bg-gray-700">
    {/* @ts-expect-error Server Component */}
    <SiteTitle size="sm" />
    ©︎ {new Date().getFullYear()} {organization}
  </footer>
);
