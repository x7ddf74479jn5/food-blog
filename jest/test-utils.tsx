import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
