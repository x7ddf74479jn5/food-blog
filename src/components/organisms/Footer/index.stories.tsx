import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import { FooterView } from "./FooterView";

export default {
  title: "organisms/Footer",
  component: FooterView,
} as ComponentMeta<typeof FooterView>;

export const Default: ComponentStoryObj<typeof FooterView> = {
  args: {
    organization: "Pandashark",
  },
};
