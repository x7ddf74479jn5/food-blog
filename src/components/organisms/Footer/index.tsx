import { fetchConfig } from "@/repositories";

import { FooterView } from "./FooterView";

export const Footer = async function () {
  const { organization } = await fetchConfig();

  return <FooterView organization={organization} />;
};
