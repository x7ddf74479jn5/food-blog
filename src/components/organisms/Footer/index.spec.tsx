import { render, screen } from "jest/test-utils";
import { mockConfig } from "mocks/data";

import Footer from ".";

describe("components/organisms/Footer", () => {
  const { organization, siteTitle } = mockConfig;

  it("OK: 表示が正しい", () => {
    render(<Footer organization={organization} siteTitle={siteTitle} />);
    const footer = screen.getByRole("contentinfo");
    expect(footer).toBeInTheDocument();
    expect(footer.tagName).toBe("FOOTER");
    const anchor = screen.getByRole("link");
    expect(anchor).toHaveTextContent(siteTitle);
    expect(anchor).toHaveAttribute("href", "/");
    expect(footer).toHaveTextContent(organization);
  });
});
