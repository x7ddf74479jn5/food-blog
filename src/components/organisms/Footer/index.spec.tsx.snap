// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/Footer snapshot 1`] = `
<footer
  className="flex flex-row gap-2 place-content-center py-2 mx-auto mt-auto w-full text-xs bg-gray-50 dark:bg-gray-700"
>
  <a
    className="text-xs font-bold text-green-500"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Food Blog
  </a>
  <span
    className=""
  >
    ©︎ 2021 
    Pandashark
  </span>
</footer>
`;
